bot/exts/help_channels/_cog.py

```diff
@@ -38,9 +38,18 @@ async def cog_load(self) -> None:
         self.help_forum_channel = self.bot.get_channel(constants.Channels.python_help)
         if not isinstance(self.help_forum_channel, discord.ForumChannel):
             raise TypeError("Channels.python_help is not a forum channel!")
+        await self.check_all_open_posts_have_close_task()
```

Comment:
While waiting for this, will the cog still be able to handle events or is the cog not considered ready yet? There are two concerns:

1. If this takes a long time, then we may miss events
2. If the cog does handle events during this, then are there any race conditions where multiple tasks may get scheduled for the same post?

Reply:
The cog isn't considered ready yet, so won't handle events.

Looking at what this does, it will only make API requests if the channel isn't cached, or if a post needs to be closed, so it shouldn't take that long in total.

The channel was actually always fetched via an API requesat here, which meant this cog ook a while to load. I just changed that in [this commit](https://github.com/python-discord/bot/pull/2959/commits/b04362048da8703554e4025e6535c628359d4bfb) in this PR.

Reply:
I'm not too concerned. Let's just try it and see if we end up with missed events - it's not catastrophic if we do.